### PR TITLE
 Adds a new structure recipe ritual to Heretic starting knowledge, "Defiled Altar", Allows them to spawn organs/bodyparts using favours gained from sacrificing.

### PR DESCRIPTION
**Pull request by Singul0**
## About The Pull Request
**Adds a new structure heretics of all paths can make, The "Defiled Altar" that is unlocked from the start for every heretic path.**
This structure allows you to calls on favours you made to your patron deity for them to grant you organs/bodyparts for you to use in rituals.

**The Recipe:**

- 1 Wooden table (built)
- 1 Sheet of cloth

Favours are gained from sucessfully sacrificing your targets. gaining two favours for every sacrifice. (four if you're a flesh heretic). The exchange rate of favours-to-bodyparts/organs is 1:1. You can view how many favours you have by looking at the heretic information tab.

Pictures:
image
image
## Why It's Good For The Game
I felt that ever since sacrificing doesn't disembowel or gib targets anymore, doing the side rituals doesn't feel worth it anymore. you're better off using the single blade and mansus you get shift start that you gradually buff throughout the round. at most probably a focus to use spells.

This leads to a (in my opinion) boring and bland gameplay where you are actively incentivized to stick to the reliable mansus and blade, where if you try to do any of the sidepaths the gameplay loop of, sacrifice --> get stronger --> sacrifice is broken by you having to spend 5+ minutes for the obscure stuff that you need.

This is a far cry from other solo antags like traitors or clings where they have an entire warehouse of variety gadgets and gizmos that they could easily obtain just at a push of a button equally. Imagine if you had to find and build your own microlasers, poison kit or emag. You'd just stick with what works and reliable (esword and ebow) instead of using the more niche and obscure items available. the time sink you need to waste to do some of these obscure sidepaths rituals, in my opinion is a huge turn-off for gameplay variety and creativity.

It breaks the gameplay flow trying to get stuff for the sidepaths, consuming precious time that you don't really have. It's not even like all of these hassles were intended. the recipes were designed from back when the sacrifices still gibbed, and I think that one way to incentivize more varied gameplay and unique scenarios is to make the recipes for doing sidepath rituals more organically sewn into the gameplay flow. Which I try to achieve in this PR

[Also refer to this forum post](https://tgstation13.org/phpBB/viewtopic.php?f=9&t=34185&p=688726#p688726)
## Changelog
:cl:
add: Added "Defiled Altar" structure, available to build for every heretic path
balance: Heretics can now get organs/bodyparts by using favours gained from sacrificing targets. 
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
I don't like this,
If you simply wanted sacrifices to produce organs then just make them do that instead of having some unecessary middle step.

Frankly speaking I also just greatly prefer Melbert's earlier vague idea of "sacrificing people should remove their organs and replace them with evil organs" but I'm not going to tell you to do that because it's a lot of work on a totally different task.
But I don't like this as a fix to this issue, it just vibes really bad to create sourceless organs out of thin air.

Also your topic link is completely fucked and doesn't lead anywhere.

**Comment by FlamingIceGolem** [comment]
> This is a far cry from other solo antags like traitors or clings where they have an entire warehouse of variety gadgets and gizmos that they could easily obtain just at a push of a button equally. Imagine if you had to find and build your own microlasers, poison kit or emag. You'd just stick with what works and reliable (esword and ebow) instead of using the more niche and obscure items available. the time sink you need to waste to do some of these obscure sidepaths rituals, in my opinion is a huge turn-off for gameplay variety and creativity.
> 
> It breaks the gameplay flow trying to get stuff for the sidepaths, consuming precious time that you don't really have. It's not even like all of these hassles were intended. the recipes were designed from back when the sacrifices still gibbed, and I think that one way to incentivize more varied gameplay and unique scenarios is to make the recipes for doing sidepath rituals more organically sewn into the gameplay flow. Which I try to achieve in this PR
> 

Nail on the head, I even made a silly meme with a similar point that I was originally going to plop in the Heretic Ascension removal PR but I didn't get around to making a proper response

Heretic (1)

But I think it would be better to just fundamentally redo how Heretic gets access to their abilities/buffs/rituals, even just getting enough wood for a table and a bedsheet for cloth to get your unique antag thing in the first place is a significant hassle that Traitor/Ling doesn't have to bear with, players tend to do more fun interesting things with Traitor/Ling due to actually having some time to pull it off. 

I think having to spend points to get access to your Rituals should be removed, since you already need to spend significant effort to just prepare them in the first place, and by the time you unlock them, you're already under severe pressure because even if you did things perfectly stealthfully because the Tears in Realities declare there are Heretics about

**Comment by the-orange-cow** [comment]
centcom doesn't want you to know this but the monkey organs in genetics are free, you can just take them home with you.  I have 500 monkey organs at home right now

**Comment by DaCoolBoss** [comment]
Thematically it feels weird to pull body parts from the Evil Dimension, just to use them as offerings for the Evil Gods of the Evil Dimension. Feels more like you're participating in some kind of magical organ market than empowering an Eldrich entity with parts of your friends and colleagues for a scrap of their power.

**Comment by MrMelbert** [comment]
> Nail on the head, I even made a silly meme with a similar point that I was originally going to plop in the Heretic Ascension removal PR but I didn't get around to making a proper response

Dog you gotta wake up

image

**Comment by Singul0** [comment]
> centcom doesn't want you to know this but the monkey organs in genetics are free, you can just take them home with you. I have 500 monkey organs at home right now

As I already outlined before, having to harvest monkeys is an uneeded timesink, especially so if your job is not affiliated with medical, genetics or botany. This timesink wasn't accounted for when heretic recipes were designed, Since back then you'd get organs/bodyparts from sacrifices anyways. You can also always print them from some protolathes, but what about the heretics with a less priveleged job. How would they easily get them?

**Comment by FlamingIceGolem** [comment]
> Dog you gotta wake up
Funny and I do want Rep requirements from weapons removed, but it's not true. Things like the Makarov/Revolver have 0 rep requirement and can be gotten immediately, and with the items that are rep-locked like E-Sword you can just sit around in a closet, jerk off for a bit then immediately get access to it once the waiting period is up. Whereas you always have to engage in a scavenger hunt first to get anything as Heretic,

**Comment by ATH1909** [comment]
> Frankly speaking I also just greatly prefer Melbert's earlier vague idea of "sacrificing people should remove their organs and replace them with evil organs" but I'm not going to tell you to do that because it's a lot of work on a totally different task. But I don't like this as a fix to this issue, it just vibes really bad to create sourceless organs out of thin air.

won't that mess with some of the stuff we've tied to organs now, like ethereal crystalization, genetics infusions, liver-based job bonuses, and food preferences?

**Comment by itseasytosee** [comment]
> > Dog you gotta wake up
> 
> jerk off to a hentai for a bit then instantly get the rep-locked weapon you want once the waiting period is up

You didn't have to say this bit.

**Comment by Deultus** [comment]
in general, this does not sound bad, but I would change it not in terms of the fact that it creates an organ, but conditionally transforms it, very often something specific like the left leg is needed for the ritual. it would be much better if it turned the conditional left arm into the right leg or any other part of the body, the liver into the eyes.

**Comment by Profakos** [comment]
This correctly identifies the problem of heretics not having time to do nonantagonist activities due to the ever increasing amount of requirements, thus making them opt out of the social deduction part of the social deduction game. However I think it offers the incorrect solution to the problem, which i think should be simplifying the recipes in some way.  Arguably, this is easier said then done. Perhaps, first instance of each item granting ritual is free? Might be a rather drastic change.

**Comment by Admiral-Gigaton** [comment]
good pr, love it. but you are just taking from the mansus, and then handing it back, pretty much. "i need eyes for this ritual of knowledge! hey mansus, can i get some? thanks! hey mansus, i got the eys n shit you needed for the ritual! thanks!" someone else mentioned the turn organs into other organs bit, i like that. only issue is "flashlight eyes speedrun"

File: code/modules/antagonists/heretic/structures/defiled_altar.dm [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+	switch(picked_type)
+		if("organs")
+			for(var/obj/item/organ/organs as anything in organs_list)
+				names_to_path[initial(organs.name)] = organs
+				choices[initial(organs.name)] = image(icon = initial(organs.icon), icon_state = initial(organs.icon_state))
+		if("bodyparts")
+			for(var/obj/item/bodypart/bodyparts as anything in bodyparts_list)
+				names_to_path[initial(bodyparts.name)] = bodyparts
+				choices[initial(bodyparts.name)] = image(icon = initial(bodyparts.icon), icon_state = initial(bodyparts.icon_state))
```
Comment (LemonInTheDark) [inline code/modules/antagonists/heretic/structures/defiled_altar.dm:110]: Rather then writing code to loop over a list of type options twice, why not use this switch to decide which list to use, and then loop over that afterwards? cleaner

**Comment by Singul0** [comment]
> won't that mess with some of the stuff we've tied to organs now, like ethereal crystalization, genetics infusions, liver-based job bonuses, and food preferences?

that was like three fourths of my reason not to go that direction. way before this, I tried making it so that it spews off a copy of the organs but it was a hacked together job

**Comment by Jacquerel** [comment]
I would probably make whatever the heretic effect we want to be a component applied to organs which also changes their colour or something

**Comment by Jacquerel** [comment]
As I said before I agree that there's a problem here but I don't think this is a good suggestion from _either_ a mechanical or flavour angle. 
Performing a middle step ritual to produce organs from mid-air only to seemingly return them to the being that sent them to you is both an unecessary amount of extra busy work and also kind of weird, especially when it comes wth an extra structure that can be used to incriminate you.

Frankly if you just want extra organs but think that the more complicated "heretic curse" solution is too tricky for you, just having sacrifices spit out random organs without removing them from the sacrificee would solve most of these problems without needing the extra shrine step, but I don't think the shrine ritual concept is a good fix for this problem.

**Pull request closed without merging**